### ISSUE TITLE
Relative paths in configurations provided on the command line are relative to main configuration

### ISSUE THREAD

**Issue by jackwilsdon** [sampsyo/beets#1594]
For example, if I create a `my_config.yaml` in a directory named `foo` with the following settings:

```yaml
directory: Music
```

And load it using `beets -c`, it uses the directory `~/.config/beets/Music` instead of `foo/Music`.

**Comment by sampsyo** [comment]
Yes, that's the case—it's definitely somewhat confusing, but it's this way because there's no other way to name paths relative to "the beets configuration directory" (which may, for example, be set with an environment variable). Relative paths like that let you change the name, for example, of the [state file](https://github.com/sampsyo/beets/blob/master/beets/config_default.yaml#L77) while leaving it in the configuration directory.

Anyway, maybe we can make this less confusing without breaking too many eggs. But at the moment, this is [the stated behavior in Confit](https://github.com/sampsyo/confit/blob/master/confit.py#L1244-L1248).

**Comment by jackwilsdon** [comment]
Would it be possible to add some form of variable expansion to Confit? So you could do something like this:

```yaml
directory: %(CONFIG_DIRECTORY)/Music
```

Where `%(CONFIG_DIRECTORY)` expands to the path of the config file being loaded?

**Comment by sampsyo** [comment]
Hmm; I'd love to avoid adding general variable expansion---that feature has made other configurations way too complex and surprisingly difficult to maintain. But maybe there's some specific point solution that makes it natural to refer both to in-config-directory files and custom-config-relative paths. Perhaps a good way to start would be to enumerate the use cases where you would want one or the other so we can identify patterns that suggest a specific design.

**Comment by jackwilsdon** [comment]
Here's my current predicament:
I own a MacBook and a Windows machine and I synchronise the following directory using git-annex:
```
Music
|- Music Library
   |- *all of my music as laid out by beets*
|- Config.yml
|- Library.db
```
In my Config.yml, I'd like to be able to specify "use library relative to this configuration". This means I can have my beets library and config (the repository) on Windows in a different place (e.g. `%USERPROFILE/Documents/Music/Music Library`) than on my Mac (e.g. `~/Music/Music Library`) (the `Music` directory is as shown above). I could have two configuration files (one for each platform) but then I'd need to keep both up to date with one another if I change my beets configuration.

**Comment by sampsyo** [comment]
Got it; that's a nice setup.

I still think it could be totally defensible to switch to a policy where paths in config files are relative to the config. It certainly sounds more intuitive than our current strategy on first glance! We'd just need to audit everything to be sure we're not missing any cases where the other thing is clearly what you want.

**Comment by jackwilsdon** [comment]
I agree it'd be nice to see how other people use their configurations so that we can gauge whether or not it's a good move to change how the paths work.

**Comment by sampsyo** [comment]
Fortunately, my guess is that relative paths are relatively (sorry) uncommon—but yes, I'd also love to hear examples.

**Comment by sampsyo** [comment]
Alright; having heard no objection, I'm going to label this as a feature request. Seems fine to make this policy change whenever we want.
